xul/bootstrap.js

```diff
@@ -0,0 +1,308 @@
+    let citem = this.browserWindow.document.getElementById("esrc-explorer-item");
+    cmenu.removeChild(citem);
+    this.targetUrlMatchPattern = null;
+  },
+};
+
+function BrowserWindowObserver(aHandlers) {
+  this.handlers = aHandlers;
+ }
+
+BrowserWindowObserver.prototype = {
+  observe: function(aSubject, aTopic, aData) {
+    if (aTopic == "domwindowopened") {
+      aSubject.QueryInterface(Ci.nsIDOMWindow).addEventListener("load", this, false);
+    } else if (aTopic == "domwindowclosed") {
+      if (aSubject.document.documentElement.getAttribute("windowtype") == "navigator:browser") {
+        this.handlers.onShutdown(aSubject);
+      }
+    }
+  },
+  handleEvent: function(aEvent) {
+    let aWindow = aEvent.currentTarget;
+    aWindow.removeEventListener(aEvent.type, this, false);
+
+    if (aWindow.document.documentElement.getAttribute("windowtype") == "navigator:browser") {
+      this.handlers.onStartup(aWindow);
+    }
+  }
+};
+
+function browserWindowStartup(aWindow) {
+  navigator = aWindow.navigator;
+  aWindow.gBrowser || aWindow.getBrowser();
+  aWindow.gBrowser.ESrcExplorer = new ESrcExplorer(aWindow);
+}
+
+function browserWindowShutdown(aWindow) {
+  aWindow.gBrowser.ESrcExplorer.done();
+  delete aWindow.gBrowser.ESrcExplorer;
+}
+
+var esrcexplorerObserver = {
+  observe: function(aSubject, aTopic, aData) {
+    if (aData == "Run") {
+      let mrw = Services.wm.getMostRecentWindow("navigator:browser");
+      let newtab = mrw.gBrowser.addTab("chrome://esrc-explorer/content/crxviewer.html", {relatedToCurrent: true})
+      mrw.gBrowser.moveTabTo(newtab, mrw.gBrowser.tabContainer.selectedIndex + 1);
+      mrw.gBrowser.selectedTab = newtab;
+    }
+  }
+};
+
+function startup(aData, aReason) {
+  let defaultBranch = Services.prefs.getDefaultBranch(branch);
+  defaultBranch.setCharPref("simpleStorage", "{}");
+  defaultBranch.setBoolPref("showButton", true);
+  defaultBranch.setBoolPref("showContext", true);
+
+  Cu.import("chrome://esrc-explorer/content/MatchPattern.jsm");
+
+  Services.scriptloader.loadSubScript("chrome://esrc-explorer/content/chrome-platform-info.js");
+  Services.scriptloader.loadSubScript("chrome://esrc-explorer/content/cws_pattern.js");
+
+  if (!styleSheetService.sheetRegistered(styleSheetURI, styleSheetService.USER_SHEET)) {
+    styleSheetService.loadAndRegisterSheet(styleSheetURI, styleSheetService.USER_SHEET);
+  }
+
+  Services.obs.addObserver(esrcexplorerObserver, "esrcexplorerEvent", false);
+
+  gWindowListener = new BrowserWindowObserver({
+    onStartup: browserWindowStartup,
+    onShutdown: browserWindowShutdown
+  });
+  Services.ww.registerNotification(gWindowListener);
+
+  let winenu = Services.wm.getEnumerator("navigator:browser");
+  while (winenu.hasMoreElements()) {
+    browserWindowStartup(winenu.getNext());
+  }
+}
+
+function shutdown(aData, aReason) {
+  if (aReason == APP_SHUTDOWN) return;
+
+  Services.ww.unregisterNotification(gWindowListener);
```

Comment:
`BrowserWindowObserver` also registers a `load` event, but that is never removed.